### ISSUE TITLE
Slowness in KD with Nemo RL during nemotron training

### ISSUE THREAD

**Issue by snowmanwwg** [NVIDIA-NeMo/RL#2490]
Off-policy KD is much slower per step. Main reasons (to-dos)
- need to make sure the ray driver doesn’t OOM on large K and context length when passing logits from teacher to student
- pipeline the teacher inference with student training
@yfw  can add more details.
